internal/launcher/launcher.go

```diff
@@ -8,35 +8,44 @@ import (
 	"go.k6.io/k6/internal/build"
 )
 
-// Launcher is a k6 Launcher
-// It analyses the requirements of a k6 execution. If required, provisions a k6Runner to satisfy these
-// requirements.
+// commandExecutor executes the requested k6 command line command.
+// It abstract the execution path from the concrete binary.
+type commandExecutor interface {
+	run(*state.GlobalState)
+}
+
+// Launcher is a k6 launcher. It analyses the requirements of a k6 execution,
+// then if required, it provisions a binary executor to satisfy the requirements.
 type Launcher struct {
+	// gs is the global state of k6.
 	gs *state.GlobalState
-	// provision function receives a list of dependencies with their constrains and returns
-	// a k6runner than satisfies them
-	provision func(*state.GlobalState, k6deps.Dependencies) (k6Runner, error)
-	// k6Runner to execute k6 command
-	runner k6Runner
+
+	// provision generates a custom binary from the received list of dependencies
+	// with their constrains, and it returns an executor that satisfies them.
+	provision func(*state.GlobalState, k6deps.Dependencies) (commandExecutor, error)
+
+	// commandExecutor executes the requested k6 command line command
+	commandExecutor commandExecutor
 }
 
 // New creates a new Launcher from a GlobalState using the default fallback and provision functions
 func New(gs *state.GlobalState) *Launcher {
+	defaultRunner := &currentBinary{}
```

Comment:
I'm not sure what nit Pablo referred to @codebien, but I think this variable name is likely the only one missing the renaming. I'd advocate for "defaultExecutor" or "defaultBinary", or something along those lines as we no longer use "runner".

Btw, thanks for this renaming, now I like how it ended up, not only around the "executor" vs "runner", but also the other part of the rename, like "customBinary", etc. 🙇🏻

Reply:
@joanlopez was the same you pointed out. It's Strange, I was sure I made the comment in the code.